node/test/error_backoff.js

```diff
@@ -0,0 +1,204 @@
+
+    backoff.handleError({
+        type: 'tchannel.timeout'
+    }, 'bob', null);
+    assert.equal(Object.keys(backoff.reqErrors).length, 0, 'nothing should be added when serviceName is invalid');
+
+    assert.ok(!backoff.getBackoffError(null, 'steve'), 'should not backoff on invalid cn');
+    assert.ok(!backoff.getBackoffError('steve', null), 'should not backoff on invalid serviceName');
+
+    assert.end();
+});
+
+allocCluster.test('should not backoff on good traffic', {
+    numPeers: 2,
+    channelOptions: {
+        timers: timers
+    }
+}, function t(cluster, assert) {
+    var one = cluster.channels[0];
+    var two = cluster.channels[1];
+    var sub = one.makeSubChannel({
+        serviceName: 'server'
+    });
+
+    sub.register('/normal-proxy', normalProxy);
+
+    var twoSub = two.makeSubChannel({
+        serviceName: 'server',
+        peers: [one.hostPort]
+    });
+
+    send(twoSub, '/normal-proxy', null, onResp);
+    function onResp(empty, err, res) {
+        assert.ifError(err);
+        assert.equal(Object.keys(two.errorBackoff.reqErrors).length, 0, 'nothing should be added');
+        assert.end();
+    }
+
+    function normalProxy(req, res, arg2, arg3) {
+        res.headers.as = 'raw';
+        res.sendOk(arg2, arg3);
+    }
+});
+
+allocCluster.test('backoff on errors', {
+    numPeers: 2,
+    channelOptions: {
+        timers: timers,
+        backoffRate: 2
+    }
+}, function t(cluster, assert) {
+    var one = cluster.channels[0];
+    var two = cluster.channels[1];
+    var sub = one.makeSubChannel({
+        serviceName: 'server'
+    });
+
+    sub.register('/normal-proxy', normalProxy);
+    sub.register('/busy', busy);
+
+    var twoSub = two.makeSubChannel({
+        serviceName: 'server',
+        peers: [one.hostPort]
+    });
+
+    series([
+        send.bind(null, twoSub, '/busy', check1),
+        send.bind(null, twoSub, '/busy', check2),
+        send.bind(null, twoSub, '/normal-proxy', check2),
+        send.bind(null, twoSub, '/normal-proxy', check3),
+    ], function done() {
+        assert.end();
+    });
+
+    function normalProxy(req, res, arg2, arg3) {
+        res.headers.as = 'raw';
+        res.sendOk(arg2, arg3);
+    }
+    function busy(req, res, arg2, arg) {
+        res.headers.as = 'raw';
+        res.sendError('Busy', 'too busy!');
+    }
+    function check1(err, res) {
+        assert.ok(err, 'there should be an error');
+        assert.equal(err.type, 'tchannel.busy', 'error type as expected');
+    }
+    function check2(err, res) {
+        assert.ok(err, 'there should be a backoff error');
+        assert.equal(err.type, 'tchannel.busy', 'backoff error type as expected');
+        assert.equal(err.subType, 'tchannel.backoff.error', 'backoff error subType as expected');
+    }
+    function check3(err, res) {
+        assert.ifError(err, 'there should be no error');
```

Comment:
I encourage everyone to use the pattern: `if (err) return assert.end(err);` instead of this err and continue pattern.